api/gateway.proto

```diff
@@ -131,7 +131,7 @@ message Gateway {
   // Encrypted Gateway Secrets.
   // This value cannot be set/retrieved in Gateway API calls. Use the StoreGatewaySecret, RetrieveGatewaySecret requests.
   // It can however be deleted by DeleteGateway since it's part of the entity.
-  Secrets secrets = 22;
+  map<string,Secret> secrets = 22 [(validate.rules).map.keys.string = {pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$" , max_len: 36}];
```

Comment:
It feels a bit weird to add this field to to our public API when it can't be read or written.

Do we really need the separate RPCs for adding/retrieving gateway secrets?
Can we not do the rights check on the field mask in the regular Get/Update RPCs?
Can we not use validation to only allow certain keys to be stored?